### PR TITLE
Switched to elasticsearch-dsl

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/etalab/data.gouv.fr/issues/651
fix https://github.com/etalab/data.gouv.fr/issues/644

### CODE REVIEW

File: app/presentation/seed.py [hunk 1]
```diff
@@ -122,17 +153,106 @@ def seed_db(search_client: SearchClient = Provide[Container.search_client]) -> N
...
+        df['metric.views'] = df['metric.views'].astype(float)
+        df['metric.followers'] = df['metric.followers'].astype(float)
+        df['orga_followers'] = df['orga_followers'].astype(float)
+
+        mvbins = [-1, 0, 50, 500, 5000, df['metric.views'].max()]
+        df['reuse_views'] = pd.cut(df['metric.views'], mvbins, labels=list(range(1, 6)))
+        mfbins = [-1, 0, 2, 10, 40, df['metric.followers'].max()]
+        df['reuse_followers'] = pd.cut(df['metric.followers'], mfbins, labels=list(range(1, 6)))
+        fobins = [-1, 0, 10, 50, 100, df['orga_followers'].max()]
+        df['orga_followers'] = pd.cut(df['orga_followers'], fobins, labels=list(range(1, 6)))
```
Comment (maudetes) [inline app/presentation/seed.py:203]: I'm not sure that we should keep the same normalization as the the one for datasets, but without quantitative data to test, we can't know what would be best.

File: app/presentation/seed.py [hunk 2]
```diff
@@ -7,28 +7,58 @@
...
+            'metric.followers': 'orga_followers',
+            'metric.datasets': 'orga_datasets'
+        })
+
+        dfo['orga_followers'] = dfo['orga_followers'].astype(float)
+        dfo['orga_datasets'] = dfo['orga_datasets'].astype(float)
 
-    click.echo("Downloading catalogs.")
+        mfbins = [-1, 0, 2, 10, 40, dfo['orga_datasets'].max()]
+        dfo['orga_datasets'] = pd.cut(dfo['orga_datasets'], mfbins, labels=list(range(1, 6)))
```
Comment (maudetes) [inline app/presentation/seed.py:41]: Do we need this info for search?

File: app/containers.py [hunk 3]
```diff
@@ -8,11 +8,20 @@ class Container(containers.DeclarativeContainer):
 
     search_client = providers.Singleton(
         ElasticClient,
-        url=config.elasticsearch_url,
-        search_synonyms=config.search_synonyms
```
Comment (maudetes) [inline app/containers.py:12]: Why don't we need `search_synonyms` anymore here?

**Review by maudetes — COMMENTED** [review]
Nice, it seems to be working nicely already !

I think we're lacking `elasticsearch_dsl` in requirements.

File: app/domain/services.py [hunk 4]
```diff
@@ -21,8 +47,35 @@ def search(self, search_text: str, page: int, page_size: int) -> Tuple[list[Data
             offset = 0
 
         results_number, results = self.search_client.query_datasets(search_text, offset, page_size)
+        results = [Dataset(**hit) for hit in results]
```
[the remark this answers is not in the document]
Comment (quaxsze) [reply app/domain/services.py:50]: The search client used to return a `Dataset` entity but is now returning a dict, the dataset service is managing the instantiation of the entity

**Pull request merged**